js/base/Exchange.js

```diff
@@ -2361,36 +2361,33 @@ module.exports = class Exchange {
         return this.filterBySymbolSinceLimit (sorted, symbol, since, limit);
     }
 
-    isPostOnly (type, timeInForce = undefined, exchangeSpecificOption = undefined, params = {}) {
+    isPostOnly (type, timeInForce, params = {}) {
         /**
          * @ignore
          * @method
          * @param {string} type Order type
          * @param {string} timeInForce
-         * @param {boolean} exchangeSpecificOption True if the exchange specific post only setting is set
          * @param {dict} params Exchange specific params
          * @returns {boolean} true if a post only order, false otherwise
          */
         let postOnly = this.safeValue2 (params, 'postOnly', 'post_only', false);
-        params = this.omit (params, [ 'post_only', 'postOnly' ]);
-        const timeInForceUpper = (timeInForce !== undefined) ? timeInForce.toUpperCase () : undefined;
         const typeLower = type.toLowerCase ();
-        const ioc = timeInForceUpper === 'IOC';
-        const fok = timeInForceUpper === 'FOK';
-        const timeInForcePostOnly = timeInForceUpper === 'PO';
+        // we assume timeInForce is uppercase from safeStringUpper (params, 'timeInForce')
```

Comment:
This assumes `safeStringUpper` is always used, which it might not be, so this is error prone